src/twisted/web/static.py

```diff
@@ -290,15 +289,15 @@ def getChild(self, path, request):
             accessed.
         @rtype: An object that provides L{resource.IResource}.
         """
-        if isinstance(path, bytes):
-            try:
-                # Request calls urllib.unquote on each path segment,
-                # leaving us with raw bytes.
-                path = path.decode('utf-8')
-            except UnicodeDecodeError:
-                log.err(None,
-                        "Could not decode path segment as utf-8: %r" % (path,))
-                return self.childNotFound
+        if not isinstance(path, bytes):
+            raise TypeError(
+                "{} is type: {}, not bytes".format(path, type(path)))
+        try:
+            path.decode('utf-8')
```

Comment:
You need to use the decoded path: https://github.com/twisted/twisted/pull/894#discussion_r143305534

Reply:
I found that when I used the decoded path here, it didn't work due to cascading dependencies in the call chain of this function, and I needed to keep it as bytes.